agent/structs/config_entry_gateways.go

```diff
@@ -526,10 +561,10 @@ func (e *TerminatingGatewayConfigEntry) Validate() error {
 		}
 
 		// Check for duplicates within the entry
-		if ok := seen[cid]; ok {
+		if ok := serviceSet[cid]; ok {
```

Comment:
I think we should do the same validation for duplicate endpoints, because we are doing a separate set of intentions for endpoints I don't think that a service and an endpoint with the same name is an issue though

Reply:
I agree with your point but I create a new map for endpoints to check for duplicates ([line 578](https://github.com/hashicorp/consul/pull/12947/files#diff-26c007c7c0d5e30fabd316255cc13ac3ffcfac5295ae8a66bcca29c197054664R578)). Could you be more specific?